### PR TITLE
fix: codegen headless fix & ask AWS profile info when info not present when checking out a new env

### PR DESCRIPTION
**Pull request by kaustavghosh06**
Codegen headless fix & ask AWS profile info when info is not present when checking out a new env

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/commands/env/checkout.js [hunk 1]
```diff
@@ -22,12 +24,40 @@ module.exports = {
...
+        allEnvs[envName][provider],
+      ));
+    });
+
+    await sequential(initializationTasks);
+
+    const onInitSuccessfulTasks = [];
+    context.exeInfo.projectConfig.providers.forEach((provider) => {
+      const providerModule = require(providerPlugins[provider]);
+      onInitSuccessfulTasks.push(() => providerModule.onInitSuccessful(
```
Comment (yuth) [inline packages/amplify-cli/src/commands/env/checkout.js]: may be name it `afterInit`

Comment (yuth) [reply packages/amplify-cli/src/commands/env/checkout.js]: How is this related to codegen headless?

Comment (kaustavghosh06) [reply packages/amplify-cli/src/commands/env/checkout.js]: Take a look at the PR heading - fix: codegen headless fix & ask AWS profile info when info not present when checking out a new env

Comment (kaustavghosh06) [reply packages/amplify-cli/src/commands/env/checkout.js]: Also onInitSuccessful is a function defined in providers and it’s used in multiple places - also in the provider from the beginning - not something introduced in this PR

Comment (yuth) [reply packages/amplify-cli/src/commands/env/checkout.js]: May be these should be 2 sepearate PRs then. Just so that we can create the right change log

**Pull request closed without merging**